backend/backend/Controllers/VideosController.cs

```diff
@@ -33,9 +37,77 @@ public VideosController(BackendContext context)
         }
 
         [HttpGet,Route("all")]
-        public ActionResult<List<Video>> GetAllVideos()
+        public ActionResult<List<VideoDto>> AllVideos()
         {
-            return _db.Videos.ToList();
+            var userIdClaim = User.FindFirst(x => x.Type == ClaimTypes.NameIdentifier);
+            if (userIdClaim is null)
+            {
+                return NotFound();
+            }
+
+            if (!Guid.TryParse(userIdClaim.Value, out var userId))
+            {
+                return NotFound();
+            }
+
+            List<Video> videos = _db.Videos.Where(video => video.DeleteDate == null && video.UserId == userId).ToList();
+
+            List<VideoDto> videosDto = new List<VideoDto>();
+            videos.ForEach(v =>
+            {
+                videosDto.Add(new VideoDto()
+                {
+                    Id = v.Id,
+                    Title = v.Title,
+                    Size = v.Size,
+                    UploadDate = v.UploadDate.ToString("yyyy-MM-dd")
+                });
+            });
+
+            return Ok(videosDto);
```

Comment:
Just a suggestion, feel free to ignore it, but maybe it would be better to return 404 if `videosDto` list is empty?

Reply:
Nope, it fucks up the flow that @kibartas  has implemented, where it works with empty arrays